### ISSUE TITLE
$project does not work with boolean aggregation opperators

### ISSUE THREAD

**Issue by arhimmel** [fakemongo/fongo#280]
it seems to be with version 2.1.0 boolean operators in the project stage do not seem to have the correct behavior.  in the project stage i would expect the operators to the boolean operations on the array yet they just change the way the output looks.

```
        @Test
	public void projectAndOr() throws Exception {
		Fongo fongo = new Fongo("server1");
		FongoDB testCase = fongo.getDB("testDb");
		DBCollection collection = testCase.getCollection("testCollection");

		String val = UUID.randomUUID().toString();
		collection.insert(new BasicDBObject("boolField", true).append("d", val));
		collection.insert(new BasicDBObject("boolField", true).append("d", val));
		collection.insert(new BasicDBObject("boolField", false).append("d", val));


		List<DBObject> pipeline = new ArrayList<>();

		pipeline.add(new BasicDBObject("$group", new BasicDBObject("boolArray", new BasicDBObject("$push", "$boolField")).append("_id", "$d")));
		pipeline.add(new BasicDBObject("$project", new BasicDBObject("boolArrayOr", new BasicDBObject("$or", "$boolArray"))
				.append("boolArrayAnd", new BasicDBObject("$and", "$boolArray"))
				.append("boolArray", 1)));

		logger.info("query " + pipeline);
		AggregationOutput aggregate = collection.aggregate(pipeline);

		Iterator<DBObject> iterator = aggregate.results().iterator();

		List<DBObject> list = new ArrayList<>();
		while (iterator.hasNext()) {
			DBObject current = iterator.next();
			list.add(current);
		}

		assertEquals(1, list.size());
		assertEquals(true, list.get(0).get("boolArrayOr"));
		assertEquals(false, list.get(0).get("boolArrayAnd"));

	}
```

**Comment by arhimmel** [comment]
created a basic pr to handle these cases

**Comment by arhimmel** [comment]
fixed by #281
